internal/oobmigration/upgrade.go

```diff
@@ -37,8 +37,9 @@ func scheduleUpgrade(from, to Version, migrations []yamlMigration) ([]MigrationI
 	// over the intervals:
 	//
 	//   (1) Order intervals by increasing upper bound
-	//   (2) For each interval, choose a new version equal to the interval's upper bound if
-	//       no previously chosen version falls within the interval.
+	//   (2) For each interval, choose a new version equal to one version prior to the interval's
+	//       upper bound if (the last version prior to its deprecation) no previously chosen version
```

Comment:
```suggestion
	//       upper bound (the last version prior to its deprecation) if no previously chosen version
```

if the parenthetical refers to "interval's upper bound"